app/assets/javascripts/3dbio_viewer/src/webapp/components/protvista/ProtvistaPdbValidation.tsx

```diff
@@ -5,24 +5,83 @@ import { ProtvistaPdb, ProtvistaPdbProps } from "./ProtvistaPdb";
 declare global {
     const d3: typeof d3Module;
 }
+const data = [
+    {
+        name: "foo",
+        units: 32,
+    },
+    {
+        name: "bar",
+        units: 67,
+    },
+    {
+        name: "baz",
+        units: 81,
+    },
+    {
+        name: "hoge",
+        units: 38,
+    },
+    {
+        name: "plyo",
+        units: 28,
+    },
+    {
+        name: "hogera",
+        units: 59,
+    },
+];
+
+const dimensions = {
+    width: 450,
+    height: 300,
+};
 
 export const ProtvistaPdbValidation: React.FC<ProtvistaPdbProps> = React.memo(props => {
     console.log("tracks", props.pdb.tracks);
 
     const svgRef = React.useRef<SVGSVGElement>(null);
 
+    const [selection, setSelection] = React.useState<null | d3Module.Selection<
+        SVGSVGElement,
+        unknown,
+        null,
+        undefined
+    >>(null);
+
+    const y = d3
+        .scaleLinear()
+        .domain([0, d3.max(data, d => d.units) || 0])
+        .range([dimensions.height, 0]);
+
+    const x = d3
+        .scaleBand()
+        .domain(data.map(d => d.name))
+        .range([0, dimensions.width])
+        .paddingInner(0.05);
+
     React.useEffect(() => {
```

Comment:
To avoid having large component blocks, let's move the content of the useEffect to a separate function (at the bottom of the file is ok)